packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -357,6 +369,33 @@ class IosProject extends XcodeBasedProject {
         }
       }
     }
+
+    // The build settings of the watchOS companion app's scheme
+    // may contain the key INFOPLIST_KEY_WKCompanionAppBundleIdentifier
+    final bool watchIdentifierFound = firstMatchInFile(xcodeProjectInfoFile, _watchAppCompanionAppIdPattern) != null;
+    if (watchIdentifierFound == false) {
+      return false;
+    }
+    for (final String scheme in schemes) {
+      final Map<String, String>? allBuildSettings = await buildSettingsForBuildInfo(
+        buildInfo,
+        deviceId: deviceId,
+        scheme: scheme,
+        isWatch: true,
+      );
+      if (allBuildSettings != null) {
+        final String? fromBuild = allBuildSettings['INFOPLIST_KEY_WKCompanionAppBundleIdentifier'];
+        if (bundleIdentifier == fromBuild) {
+          return true;
+        }
+        if (fromBuild != null && fromBuild.contains(r'$')) {
```

Comment:
```suggestion
        if (fromBuild != null && fromBuild.contains('$')) {
```

Reply:
I found that removing the `r` caused an error, because it expects an identifier or expression. If I escaped it, like `'\$'`, the linter suggests to use a raw string instead of escaping https://dart-lang.github.io/linter/lints/use_raw_strings.html. So I left it as is.